extends `plugin:import/recommended` and fix warnings

B2o5T opened the pull request:
don't be scary of modified code lines, it's mostly only `yarn.lock` changes (:

B2o5T commented on examples/cm6-graphql-legacy-parcel/package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "example-cm6-graphql-parcel",
+  "name": "example-cm6-graphql-legacy-parcel",
```
![image](https://user-images.githubusercontent.com/7361780/204138458-c01ac363-8f84-4990-96b0-24abb4f90b09.png)

B2o5T commented on .eslintrc.js:
```diff
@@ -22,7 +22,6 @@ module.exports = {
     react: {
       version: 'detect',
     },
-    'import/extensions': ['.js', '.jsx', '.ts', '.tsx'],
```
already included while extending `plugin:import/typescript` config

B2o5T commented on package.json:
```diff
@@ -6,10 +6,7 @@
-      "examples/monaco-graphql-webpack",
-      "examples/monaco-graphql-react-vite",
-      "examples/cm6-graphql-parcel"
+      "examples/*"
```
I guess it's fine to include all examples in workspace

acao replied:
we removed the examples from linting a while ago, this slows things down and isn't very valuable

B2o5T replied:
here is `workspaces` field in `package.json`, do you want to completely remove all `examples` from here to avoid installing their dependencies?

We still can keep ESLint lint for examples folder, to avoid some mistakes

B2o5T commented on package.json:
```diff
@@ -117,9 +113,9 @@
     "cors": "^2.8.5",
     "cross-env": "^7.0.2",
     "cspell": "^5.15.2",
-    "cypress": "^4.7.0",
```
moved inside `graphiql`'s package.json

B2o5T commented on packages/graphiql/package.json:
```diff
@@ -54,6 +54,7 @@
     "@graphiql/toolkit": "^0.8.0",
     "entities": "^2.0.0",
     "graphql-language-service": "^5.1.0",
+    "regenerator-runtime": "0.13.11",
```
added as dependency since imported in https://github.com/graphql/graphiql/blob/6c3ca902d769893a4201f8856299af18b7e21a1b/packages/graphiql/src/cdn.ts#L10

B2o5T replied:
removed as in this PR https://github.com/graphql/graphiql/pull/2939

B2o5T commented on packages/graphql-language-service-server/package.json:
```diff
@@ -52,6 +52,7 @@
     "nullthrows": "^1.0.0",
     "vscode-jsonrpc": "^8.0.1",
     "vscode-languageserver": "^8.0.1",
+    "vscode-languageserver-types": "^3.17.2",
```
added as dependency since imported in https://github.com/graphql/graphiql/blob/6c3ca902d769893a4201f8856299af18b7e21a1b/packages/graphql-language-service-server/src/GraphQLLanguageService.ts#L48-L52

B2o5T commented on packages/graphql-language-service-server/src/index.ts:
```diff
@@ -16,4 +16,3 @@ export { Logger } from './Logger';
 export * from './GraphQLCache';
 export * from './parseDocument';
 export * from './findGraphQLTags';
-export * from './stringToHash';
```
this export does nothing since `stringToHash` export default export only
![image](https://user-images.githubusercontent.com/7361780/204149231-455283ef-75ae-470c-a595-bd43745f35dd.png)

acao replied:
sounds good to me

B2o5T commented on packages/graphql-language-service/src/parser/onlineParser.ts:
```diff
@@ -27,10 +27,6 @@
  *
  */
 
-import {
-  LexRules as LexRulesType,
-  ParseRules as ParseRulesType,
-} from './Rules';
```
duplicated imports with on line 33

B2o5T commented on packages/monaco-graphql/package.json:
```diff
@@ -36,6 +36,7 @@
     "graphql": "^15.5.0 || ^16.0.0",
-    "monaco-editor": "^0.20.0"
+    "monaco-editor": "^0.20.0",
+    "prettier": "^2.8.0 || ^3.0.0"
```
added `prettier` as peerDeps since `prettier` is imported here https://github.com/graphql/graphiql/blob/6c3ca902d769893a4201f8856299af18b7e21a1b/packages/monaco-graphql/src/GraphQLWorker.ts#L141-L142

B2o5T commented on packages/vscode-graphql-execution/package.json:
```diff
@@ -110,6 +110,7 @@
     "graphql-tag": "2.12.6",
     "graphql-ws": "5.10.0",
     "@whatwg-node/fetch": "0.2.8",
+    "wonka": "4.0.15",
```
added in dependency since `wonka` is imported here https://github.com/graphql/graphiql/blob/6c3ca902d769893a4201f8856299af18b7e21a1b/packages/vscode-graphql-execution/src/helpers/network.ts#L6

acao replied:
Esbuild seems to bundle the `@urql` provided version. I'm wary of managing this version in case it conflicts with `@urql` at some point though we are pining. 
Perhaps we can import these methods directly from some `@urql` package now?

B2o5T replied:
after removing `wonka` and adding `eslint-disable` comment, deploy passes 🤔

B2o5T replied:
I removed wonka from dependencies, I guess after switching to pnpm we will still need to add it to the dependencies

B2o5T commented:
@acao ani idea why netlify deploy fails ? 🤔

acao commented:
@B2o5T seems to be that it's running out of build time? The build toolchain has gotten much longer now that we're using vite + webpack4 🙈

acao commented:
@B2o5T are you able to see [the netlify build logs](https://app.netlify.com/sites/graphiql-test/deploys/6383802741941300084ad132)? 
![image](https://user-images.githubusercontent.com/1368727/205378473-defccfa0-8cb6-43df-8900-564109893788.png)

B2o5T commented:
> @B2o5T are you able to see [the netlify build logs](https://app.netlify.com/sites/graphiql-test/deploys/6383802741941300084ad132)? ![image](https://user-images.githubusercontent.com/1368727/205378473-defccfa0-8cb6-43df-8900-564109893788.png)

Unfortunately not, infinity loading
<img width="1289" alt="image" src="https://user-images.githubusercontent.com/7361780/205382613-5ec12f92-64fb-4aca-b0fb-cfbe16e69c9b.png">

B2o5T commented:
> @B2o5T seems to be that it's running out of build time? The build toolchain has gotten much longer now that we're using vite + webpack4 🙈

seems due hosting of dependencies, migration to pnpm should avoid these problems in the future 🤔

jonathanawesome commented on package.json:
```diff
@@ -5,11 +5,7 @@
-      "examples/monaco-graphql-webpack",
-      "examples/monaco-graphql-react-vite",
-      "examples/cm6-graphql-parcel"
+      "packages/*"
```
Some of the examples have readme files which instruct users to `yarn` from root to install deps. We should make sure those readme files get updated if necessary.

thomasheyenbrock replied:
+1 on this, would love if we'd adjust this before merging!

TallTed commented on .eslintrc.js:
```diff
@@ -264,10 +263,10 @@ module.exports = {
 
   overrides: [
-    // Cypress plugin, global, etc only for cypress directory
+    // Cypress plugin, global, etc. only for cypress directory
```
```suggestion
    // Cypress plugin, global, etc., only for cypress directory
```

B2o5T commented:
@acao after removing examples from workspaces, deploy passes

acao commented:
> seems due hosting of dependencies, migration to pnpm should avoid these problems in the future

indeed! once we replace or upgrade webpack for bundling graphiql, we can support pnpm.

acao reviewed (approved):
Looks good to me! Thanks for doing this!

Ignoring the examples for now is more ideal for now. I'll create a follow up issue for examples

thomasheyenbrock commented on packages/graphiql/test/afterDevServer.js:
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable react-hooks/rules-of-hooks */
```
Could we only enable rules of hooks for `jsx` and `tsx` files?

B2o5T replied:
I was thinking about this but there is a situation when the custom hook is placed in separate file and doesn't contain JSX elements, so he will have `js`/`ts` extension and not be linted 😒

thomasheyenbrock replied:
Very good point, then let's leave it as is!

thomasheyenbrock reviewed (approved):
🏆

B2o5T commented:
@thomasheyenbrock updated all examples:

- removed instruction to do `yarn`/`yarn start` from the root
- moved `typescript` to `devDeps`
- replaced all relative packages version `"file:../../` with published versions since examples no longer included in workspaces
